docs/docs/code/nodejs/using-data-stores/README.md

```diff
@@ -0,0 +1,169 @@
+---
+short_description: Store and read data with data stores.
+thumbnail: https://res.cloudinary.com/pipedreamin/image/upload/v1646763735/docs/icons/icons8-database-96_iv1oup.png
+---
+
+# Using Data Stores 
+
+In Node.js (Javascript) code steps, you can also store and retrieve data within code steps without connecting a 3rd party database. **Date Stores** are very useful for tracking data between runs of a particular workflow, making sure workflows only run once per unique record or sharing data between workflows.
+
+:::warning
+This functionality is limited to only Node.js code steps at this time.
+
+Other step languages like [Python](/code/python/), [Bash](/code/bash/) and [Go](/code/go/) do not have this feature available yet.
+
+For more information on what functionality is available for those languages, please refer to their documentation.
+:::
+
+### Adding a data store to a Node.js Step
+
+By default, Node.js steps don't have access to data stores. A data store can be added to your step by adding it as a `prop`.
+
+```javascript
+export default defineComponent({
+  props: {
+    // Define that the "db" variable in our component is a data store
+    data: { type: "store" }
+  },
+  async run({ steps, $ }) {
+    // Now we can access the data store at "this.store"
+    this.data.get("email");
+  }
+});
+```
+
+:::tip
+`props` injects variables under `this` scope in components.
+
+In the above example we essentially instructed that this step needs the data store injected into the `this.store` prop. 
+:::
+
+## Using the data store
+
+Once you have defined a data store prop for your component, then you'll be able to create a new data store or use an existing one from your account.
+
+![Create a new data store or choose another one from your account for your component](https://res.cloudinary.com/pipedreamin/image/upload/v1647626951/docs/components/CleanShot_2022-03-18_at_14.08.01_2x_fyr3p4.png)
+
+## Saving data
+
+Data stores are a key-value store, you can save data within a store using the `store.set` method. The first argument is the _key_ where the data should be held, and the 2nd argument is the _value_ assigned to that key.
```

Comment:
"Data stores are a key-value store" -> "Data stores are key-value stores"

Reply:
2nd -> "second" (my style guide tells me you should always use the English "second" when referring to orderings like this)